### PR TITLE
library: Add User Info Entry

### LINKED ISSUE #346: libportal get_user_information
https://libportal.org/method.Portal.get_user_information.html
https://libportal.org/method.Portal.get_user_information_finish.html

Let's make a demo about request user information.

We can display/present the info in a nice way in the preview.

ASHPD has a cool demo for it you can get inspiration from see "Account"

### PR DESCRIPTION
**Pull request by SoNiC-HeRE**
Closes #346

### CODE REVIEW

**Comment by SoNiC-HeRE** [comment]
## Overview:
- Kept the design minimalistic
- Used promisified methods
- Commented parts of the code for better understanding

Looking for an early review

File: src/Library/demos/User Info/main.blp [hunk 1]
```diff
@@ -0,0 +1,70 @@
+using Gtk 4.0;
+using Adw 1;
+
+Adw.StatusPage {
+  title: "User Info";
```
Comment (sonnyp) [inline src/Library/demos/User Info/main.blp:5]:
```suggestion
  title: _("User Info");
```

File: src/Library/demos/User Info/main.blp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  description: _("Request user information");
```
Comment (sonnyp) [inline src/Library/demos/User Info/main.blp:6]:
```suggestion
  description: _("Request information about the user");
```

File: src/Library/demos/User Info/main.blp [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  margin-top: 48;
+
+  Box {
+    orientation: vertical;
+    halign: center;
+
+    Label {
+      label: _("What best describes your request?");
```
Comment (sonnyp) [inline src/Library/demos/User Info/main.blp:14]:
```suggestion
      label: _("What is the reason for the request?");
```

File: src/Library/demos/User Info/main.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+const avatar = workbench.builder.get_object("avatar");
+const entry = workbench.builder.get_object("entry");
+const username = workbench.builder.get_object("username");
+const display = workbench.builder.get_object("name");
+
+Gio._promisify(
+  Xdp.Portal.prototype,
+  "get_user_information",
+  "get_user_information_finish",
+);
```
Comment (sonnyp) [inline src/Library/demos/User Info/main.js:20]: can you move this below the `import`s ? (with a newline separation)

it makes it clear that it's related

File: src/Library/demos/User Info/main.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+async function onClicked() {
+  // call the portal function
```
Comment (sonnyp) [inline src/Library/demos/User Info/main.js:23]:
```suggestion
```

File: src/Library/demos/User Info/main.js [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  const reason = entry.get_text();
+  let user_info = null;
+  user_info = await portal.get_user_information(parent, reason, null, null);
+
+  //retrieve values from GVariant
+  user_info = user_info.deepUnpack();
```
Comment (sonnyp) [inline src/Library/demos/User Info/main.js:29]: I really like the gvariant description comment you made in the other PR, I would do something similar here

```suggestion
  const result = await portal.get_user_information(parent, reason, null, null);
  // result is a ...
  const user_info = user_info.deepUnpack();
```

File: src/Library/demos/User Info/main.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+  const id = user_info["id"].deepUnpack();
+  const name = user_info["name"].deepUnpack();
+  const uri = user_info["image"].deepUnpack();
+  const file = Gio.File.new_for_uri(uri);
+  const texture = Gdk.Texture.new_from_file(file);
+
+  //update values
```
Comment (sonnyp) [inline src/Library/demos/User Info/main.js:36]: this comment isn't helpful IMHO

File: src/Library/demos/User Info/main.js [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+  username.label = id;
+  display.label = name;
+  avatar.set_custom_image(texture);
+
+  //post-request updation
```
Comment (sonnyp) [inline src/Library/demos/User Info/main.js:41]: ditto

File: src/Library/demos/User Info/main.blp [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+
+      Entry entry {
+        placeholder-text: _("Reason");
+      }
+
+      Button button {
+        label: _("Submit");
+        halign: center;
+        styles ["suggested-action"]
+      }
```
Comment (sonnyp) [inline src/Library/demos/User Info/main.blp:32]: can you use a AdwEntryRow and a "Request" button like ASHPD app?

image

I would drop the question too

File: src/Library/demos/User Info/main.blp [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+      Adw.ActionRow {
+        title: _("Name :");
+
+        Label name {
+          label: _("N/A");
+        }
+      }
+
+    }
```
Comment (sonnyp) [inline src/Library/demos/User Info/main.blp:61]: Can you do like ASHPD and hide this until we have the information ?

You could use a Revealer for :sunglasses: effect

**Review by sonnyp — CHANGES_REQUESTED** [review]
Looks great

I think we should call the demo "Account" though, it's a bit clearer

**Comment by SoNiC-HeRE** [comment]
## Made the following changes:
- Updated Title to Account
- Used Revealer
- Improved UI
- Refined Code
- Kept the UI and Code minimalistic

```Used Listbox as a child since Gtk.Revealer only supports  one child widget  at a time```

**Review by sonnyp — APPROVED** [review]
Nice

**Pull request merged**